packages/grafana-ui/src/components/uPlot/MicroPlot.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { PureComponent } from 'react';
+
+import { DataFrame, getTimeField, FieldType, formattedValueToString } from '@grafana/data';
+
+import uPlot from 'uPlot/dist/uPlot.cjs';
+// import 'uPlot/src/uPlot.css';
+import { colors } from '../../utils';
+
+interface Props {
+  data: DataFrame; // assume applyFieldOverrides has been set
+  width: number;
+  height: number;
+}
+
+interface PlotState {
+  // ?? over info?
+}
+
+export class MicroPlot extends PureComponent<Props, PlotState> {
+  plot: any;
+
+  componentDidUpdate(oldProps: Props) {
+    const { width, height } = this.props;
+    if (!this.plot) {
+      console.log('NO plot yet...');
+      return;
+    }
+
+    // Update if the data changes
+    if (this.props.data !== oldProps.data) {
+      const { series, uData } = getUPlotStuff(this.props);
+      this.plot.setData(uData);
+      console.log('DATA changed!', uData, series);
+      // Assume same structure?? this.plot.setSeries(series);
+    }
+
+    if (width !== oldProps.width || height !== oldProps.height) {
+      this.plot.setSize({ width, height });
+    }
+  }
+
+  init = (element: any) => {
+    const { width, height } = this.props;
+
+    const { series, uData } = getUPlotStuff(this.props);
+    const fmtt = (val: number, dec: number) => val.toFixed(dec);
+
+    const opts = {
+      width,
+      height,
+      legend: {
+        show: false,
+      },
+      tzDate: (ts: any) => uPlot.tzDate(new Date(ts), 'Etc/UTC'),
+      scales: {
+        x: {
+          distr: 2,
```

Comment:
just a heads up - you're giving up a lot of benefits offered by `time: true` scales by doing this, such as logical temporal breakpoints and a chart that spaces ticks according to the calendar (feb should be narrower than jan). the main use case for `distr: 2` is squashing things like weekends (useful for financial, when markets are closed) by ignoring the calendar and just treating the x timestamps as if they were meaningless numbers (using divisors of 2,5,10, etc).

Reply:
Thanks!  this is my quick copy your demos and see what happens POC.  In general -- it looks super promising.  All the scale/axis configs will need to be revisited as we make it more real :)